Packages/com.unity.inputsystem/InputSystem/Plugins/OnScreen/OnScreenDeviceManager.cs

```diff
@@ -0,0 +1,43 @@
+using UnityEngine.Experimental.Input.LowLevel;
+
+namespace UnityEngine.Experimental.Input.Plugins.OnScreen
+{
+    public class OnScreenDeviceManager
+    {
+        static OnScreenDeviceManager s_Instance;
+
+        public static OnScreenDeviceManager GetOnScreenDeviceManager()
+        {
+            if (s_Instance == null)
+                s_Instance = new OnScreenDeviceManager();
+
+            return s_Instance;
+        }
+
+        public InputEventPtr GetInputEventPtrForDevice(InputDevice device)
+        {
+            InputEventPtr eventPtr;
+            var buffer = StateEvent.From(device, out eventPtr);
```

Comment:
Add argument to From that allows choosing allocator to get around problem of temp pool.